### PR TITLE
For #18608 made set a default browser functionality publicly available.

### PR DESCRIPTION
**Pull request by Amejia481**
For multiple experiments (https://github.com/mozilla-mobile/fenix/issues/18608 ,https://github.com/mozilla-mobile/fenix/issues/18376, https://github.com/mozilla-mobile/fenix/issues/18375)  we need a reusable behavior for setting a default browser, this PR make the functionality reusable. 
  
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/ext/Activity.kt [hunk 1]
```diff
@@ -48,3 +56,61 @@ fun Activity.breadcrumb(
         )
     )
 }
+
+/**
+ * Opens the set default browser setting depending on the user OS version:
+ * for devices with Q or lower takes the user to the settings page
+ * and for other allow the user to set the default browser in app.
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/ext/Activity.kt:63]: I don't think the 3 conditions are quite right from what I am reading in the code. You mean Android Q+ right for the first point?
In the second point, I think that applies to Android N+, but the else condition actually takes ppl to a sumo page. Provided some suggestion but I think we can also change this kDoc to say:
"Opens Android's Manage Default Apps Settings if possible."

Anyways, I want to be crystal that we are navigating out of the app into "Android's Manage Default Apps Settings".

```suggestion
 * - For Android Q+, navigate the user to Android's Manage Default Apps Settings
 * - For anything else, allow the user to set the default browser in app. (I think this is wrong??)
```

File: app/src/main/java/org/mozilla/fenix/ext/Activity.kt [hunk 2]
```diff
@@ -48,3 +56,61 @@ fun Activity.breadcrumb(
...
+                    SupportUtils.SumoTopic.SET_AS_DEFAULT_BROWSER
+                ),
+                newTab = true,
+                from = BrowserDirection.FromSettings
+            )
+        }
+    }
+}
+
+private fun Activity.navigateUserToDefaultAppsSettings() {
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/ext/Activity.kt:98]:
```suggestion
private fun Activity.navigateToDefaultAppsSettings() {
```

**Review by gabrielluong — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/ext/Activity.kt (hunk 2 above)
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/ext/Activity.kt:98]: Perhaps also add a kDoc for this. Can never have enough documentation to help people not have to read code.

File: app/src/main/java/org/mozilla/fenix/ext/Activity.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/ext/Activity.kt:63]: Thanks for catching this, I will go with the straightforward documentation as you suggested "**Opens Android's Manage Default Apps Settings if possible."**

File: app/src/main/java/org/mozilla/fenix/ext/Activity.kt [hunk 3]
```diff
@@ -48,3 +56,59 @@ fun Activity.breadcrumb(
         )
     )
 }
+
+/**
+ *  Opens Android's Manage Default Apps Settings if possible.
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/ext/Activity.kt:61]:
```suggestion
 * Opens Android's Manage Default Apps Settings if possible.
```

File: app/src/main/java/org/mozilla/fenix/ext/Activity.kt [hunk 4]
```diff
@@ -48,3 +56,59 @@ fun Activity.breadcrumb(
...
+                    SupportUtils.SumoTopic.SET_AS_DEFAULT_BROWSER
+                ),
+                newTab = true,
+                from = BrowserDirection.FromSettings
+            )
+        }
+    }
+}
+
+private fun Activity.navigateToDefaultBrowserAppsSettings() {
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/ext/Activity.kt:96]: Where's my kDoc?

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/ext/Activity.kt:96]: As it's private helper function I think it wasn't need it, but we can add it too.

**Pull request merged**